### PR TITLE
refactor(structured data): added str. data for articles and pages

### PR DESCRIPTION
**Pull request by dw-buildit**
Changes made to overall structured data:
- `SiteNavigationElement`s added
- Lists marked up as `ItemList`s
- `Article` markup

Other amends:
- All pages (except home and 404) use `page-intro` for their heading to get consistent spacing and layout
- Removed some job list markup in anticipation of a future gravity release (fine for merging now as those pages aren't ready yet anyway)
- `nav-card` markup extracted to it's own template (DRY)

### CODE REVIEW

File: pages/people/index.html.njk [hunk 1]
```diff
@@ -11,19 +11,20 @@ nav-index: 1
   Welcome to our new Buildit website featuring stories by and about our technical people and teams.
   News from our community, and an easy way to search the entire range of relevant job opportunities we have open.
 </p>
-<ul class="grav-c-list-cards-basic">
+<ul class="grav-c-list-cards-basic" property="schema:mainEntity" typeof="schema:ItemList">
```
Comment (james-nash) [inline pages/people/index.html.njk]: Need to ditch the `property="schema:mainEntity"` here too.

Also, if I'm not mistaken, this whole `<ul>` is identical to the one on the stories listing page. If so, would it perhaps make sense to break it out into a Nunjucks partial that both of the pages include? DRY FTW! :-)

File: pages/stories/index.html.njk [hunk 2]
```diff
@@ -6,24 +6,25 @@ schema: CollectionPage
 collection: colMainNav
 nav-index: 0
 ---
-<h1>Stories</h1>
+<h1 property="schema:name">Stories</h1>
```
Comment (james-nash) [inline pages/stories/index.html.njk]: Same comments as above on the people page apply here.

**Comment by james-nash** [comment]
@dw-buildit It looks like my previous review comments have been addressed, but I see that the list items (for stories) still have image, description, etc. props. I thought we'd cut them down to just the url to the story page, as per Google's carousel advice. Or not? Not massively bothere either way TBH.

Anyway, just wanted to check if this PR is now ready for re-review (in which case I'd probably approve it), or if there's still more work incoming and I should hold off for now.

**Comment by dw-buildit** [comment]
@james-nash Yea, most of the comments have been resolved. I haven't reinstated `CollectionPage` as a  definition on the navigation items in `layouts/default/header.njk`. Other than that I think everything was addressed as per our conversation yesterday.

As for the Carousel, I think that markup should still be fine, as ListItem allows those fields and I think it is ListItem that is being marked up there - there's no reference to Article on the list items themselves.

**Pull request merged**